x-pack/plugins/ml/public/application/services/anomaly_detector_service.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { forkJoin, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Job, JobId } from '../../../common/types/anomaly_detection_jobs';
+import { basePath } from './ml_api_service';
+import { HttpService } from './http_service';
+
+export class AnomalyDetectorService {
+  private readonly apiBasePath = basePath() + '/anomaly_detectors';
+
+  constructor(private httpService: HttpService) {}
+
+  /**
+   * Fetches a single job object
+   * @param jobId
+   */
+  getJobById$(jobId: JobId): Observable<Job> {
+    return this.httpService
+      .http$<{ count: number; jobs: Job[] }>({
+        path: `${this.apiBasePath}/${jobId}`,
+      })
+      .pipe(
+        map(response => {
+          return response.jobs[0];
+        })
+      );
+  }
+
+  getJobs$(jobIds: JobId[]): Observable<Job[]> {
+    return forkJoin(jobIds.map(jobId => this.getJobById$(jobId)));
```

Comment:
firing a loop of async requests to the server has causes us problems in the past.
The endpoint you're calling can take a comma separated list of job ids, `anomaly_detectors/job1,job2`

Reply:
Fixed in https://github.com/elastic/kibana/pull/64056/commits/17515b1cd779bf4e600e3c4eb2eb83b67014408a